MudInput: Add Accessibility Titles to Nested Icons

WalterWillis opened the pull request:
## Description
Resolves #5530 

Add Title details to the three icons nested inside of the MudInput Component for accessibility concerns in software test tools.

While performing some 508 testing, I noticed that our AutoComplete component was failing on the Clear Icon. Looks like a MudInput is being used in the source code, so I added a title value to it and the two increment/decrement icons while I was at it. 

Please let me know if there are any issues or concerns.

## How Has This Been Tested?
Unit tests run. I believe this counts as a trivial change, but I'm not certain.

## Types of changes
<!-- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)


## Checklist:
- [x] The PR is submitted to the correct branch (`dev`).
- [x] My code follows the code style of this project.
- [x] I've added relevant tests. (I believe this is a trivial change, so no tests actually added.)

WalterWillis committed:
MudInput: Add Accessibility Titles to Nested Icons

mikes-gh commented:
From what I know `title` support is inconsistent in screen readers.
`aria-label` is however defacto.
So its suggested if you need a tool tip use `title` and `aria-label`
If you don't ned a tool tip just `aria-label`

But I am open to discussion ;-)

WalterWillis commented:
That's fair. The software we use we pretty much have to follow to the letter. The people that oversee our Security and compliance are pretty strict. However, it sounds like using the title attribute along with the aria-label attribute should suffice. Is that something you agree with?

mikes-gh commented:
> That's fair. The software we use we pretty much have to follow to the letter. The people that oversee our Security and compliance are pretty strict. However, it sounds like using the title attribute along with the aria-label attribute should suffice. Is that something you agree with?

https://www.tpgi.com/using-the-html-title-attribute/

Touch cannot access title so given that a lot of apps end up on touch screens I think title as a tool tip is pretty defunct these days.  Also even on desktops people that have a hard time using a mouse may not have access to title either since keyboards often dont surface the title.  If the main reason for using title is screen readers you are better using aria-label.
Some screen readers dont read title by default because when authors use it they are not always thinking about screen readers.

Personally from a library perspective I think we should just use aria-label.

But again I'm open to discussion ;-)

WalterWillis committed:
Change Title to aria-label

Based on feedback, replaced the Title attribute with aria-label for better general usecases.

WalterWillis commented:
> https://www.tpgi.com/using-the-html-title-attribute/
> 
> Personally from a library perspective I think we should just use aria-label.

This is very useful knowledge and I agree. The compliancy standard we use is out of date and the recommendations appear to be as well. I did some testing and aria-label does satisfy our standard. It is definitely the superior fix. 

Committed and ready for review. Please let me know what you think about the label language itself, if you don't mind.

henon commented:
Thanks
